app/controllers/api/base_controller.rb

```diff
@@ -187,7 +187,11 @@ def authorize
         return false
       end
 
-      unless authenticate
+      if authenticate
+        if User.current.disabled
+          render_error('custom_error', :status => :unauthorized, locals: {message: _('User is disabled')})
```

Comment:
I think the message could be more descriptive, something like `User account is disabled, please contact administrator`.